### PR TITLE
Refine RunnerPool CRD

### PR DESCRIPTION
**Pull request by masa213f**
1. Slack notification settings

By this PR, users can write slack notification settings as follows.
And the Slack notification become optional. See the user_manual.md

```
apiVersion: meows.cybozu.com/v1alpha1
kind: RunnerPool
metadata:
  name: runnerpool-sample
spec:
  repository: "<Owner>/<your Repository>"
  replicas: 3
  slackNotification:
    enable: true
    channel: "#<channel_name>"
    extendDuration: "30s"
```

2. Add `.spec.organization` field

We can now create org-level runners with the following definition.

```
apiVersion: meows.cybozu.com/v1alpha1
kind: RunnerPool
metadata:
  name: runnerpool-sample
spec:
  organization: "<your Organization>"
  replicas: 3
```

Signed-off-by: Masayuki Ishii <masa213f@gmail.com>

### CODE REVIEW

File: docs/user-manual.md [hunk 1]
```diff
@@ -18,27 +18,32 @@ So make the `meows` namespace to prepare.
...
 ```bash
-$ GITHUB_ORG=<your Organization>
 $ kubectl create configmap meows-cm -n meows \
-    --from-literal=organization=${GITHUB_ORG}
+    --from-literal=organization-rule='^neco-test$' \
+    --from-literal=repository-rule='^neco-test/.*' \
 ```
 
-After that deploy the controller.
+Both `organization-rule` and `repository-rul` accepts golang's regular expressions.
```
Comment (kmdkuk) [inline docs/user-manual.md:32]: typo
```suggestion
Both `organization-rule` and `repository-rule` accepts golang's regular expressions.
```

Comment (masa213f) [reply docs/user-manual.md]: Great job!

File: api/v1alpha1/runnerpool_types.go [hunk 2]
```diff
@@ -58,16 +58,27 @@ type RunnerPoolSpec struct {
...
+	Slack SlackConfig `json:"slack,omitempty"`
+
+	// Extension time.
+	// If this field is omitted, users cannot extend the runner pods.
+	// +optional
+	ExtendDuration string `json:"extendDuration,omitempty"`
+}
+
+type SlackConfig struct {
 	/// Flag to toggle Slack notifications sends or not.
```
Comment (kmdkuk) [inline api/v1alpha1/runnerpool_types.go:87]: `/` is one too many.

**Review by kmdkuk — COMMENTED** [review]
Other than the points I commented on, it looks good.

File: api/v1alpha1/runnerpool_types.go (hunk 2 above)
Comment (masa213f) [reply api/v1alpha1/runnerpool_types.go]: Thank you!

**Review by kmdkuk — APPROVED** [review]
LGTM

**Pull request merged**